### PR TITLE
Merge b61a8729

### PR DESCRIPTION
**Pull request by shaleh**
Conflicts:
    - src/dosfns.c -- we remove it
    - src/fileio.c -- conflict over DOS ifdef changes
    - src/lread.c -- conflict because readevalloop is no longer static
    - src/minibuf.c -- `minibuffer-completion-contents` removed and we had ported it
    - src/process.c -- ifdef subprocesses no longer needed due to DOS changes
    - src/w16select.c -- we removed it
    - src/xfns.c -- X and GTK ifdef changes
    - src/xmenu.c -- X and GTK ifdef changes
    - test/lisp/net/tramp-tests.el -- windows checks

b61a872926 * lisp/calendar/diary-lib.el (diary-remind): Fix bug#30455.
82379efaaf ; Add test for iter-lambda variable shadowing (Bug#26073)
298275c595 ; Replace "Written by" with standard "Author:" header in some C files
4bee1b8bac Stop keeping multiple doc copies for items defined multiple times
6c7186de3b Doc tweak for x-use-underline-position-properties
98f4e336e8 Merge from origin/emacs-26
333d6f4d99 More changes in the Emacs manual
52ca0d1a3e * lisp/vc/vc.el (vc-deduce-backend): Use ignore-errors.
69e804669b Don't signal error in vc-deduce-backend
1d135af78c Fix NS-related compilation errors.
f568c916b6 * doc/misc/tramp.texi: Index more user options.
92ca881dc8 Minor change in the Emacs manual
e055a12839 NEWS update about 'string-trim'
4b70de228d Expand direct color description in Emacs FAQ
82c3be8347 * lisp/gnus/message.el (message-ecomplete-capf): Fix call convention.
0676633e8f * lisp/progmodes/python.el: Bump package version to 0.26.
b8c4151808 Work if libpng is in /usr/local (2nd try)
96b6e243ec Clarify TRAMP process-environment interaction (Bug#30419)
4fa467eefa * lisp/progmodes/grep.el (grep-num-matches-found): New variable.
3a718ffca0 Finish renaming to internal--daemon-sockname
99cf1bb011 Rename internal--external-sockname and document (Bug#24218)
b9a3cd6899 Update `server-name' :version & document sockname (Bug#24218)
d43a724f4e Minor cleanups for server-name fix (Bug#24218)
e1ca0ea872 Fix `server-name' and `server-socket-dir' for (Bug#24218)
a22820a31c Avoid aborts in cm.c due to too small TTY frame
b3f45140ec Minor fixes to .gdbinit
26f64410fd Another set of improvements in the Emacs manual
1590f8823d Revert libpng /usr/local patch
cef3b4248c Fix help in mode-line-mode-menu
e9ca57cfcb Work even if libpng is installed in /usr/local
76a1bd1896 Fix an infloop in python-nav-beginning-of-statement
37b2f9f356 Merge from pkg-config
65ebf78909 Fix modifier keys on NS port
68d8925077 Tidy up NS keybindings
f7a853d818 Remove NS special handling of 'frame-title-format'
f8a493c800 Improve documentation of desktop restoring
5a75cd1f8f * src/xdisp.c (Vmessage_stack): Fix comment to match code.
10637af91a Improve the Emacs manual's chapter "Frames"
13f4b518d0 Fix handling of file notifications in tramp-gvfs.el
d924953e05 Fix unbound mm-uu-entry in mm-uu
09465bfa06 * lisp/window.el (display-buffer--maybe-at-bottom): New function (bug#30314).
aaeb101d89 * lisp/comint.el (comint-after-pmark-p): Check if buffer has a live process.
cb2487b173 Improve Emacs user manual in fixit.texi
7f6153d956 Support standard Terminfo direct mode terminals
66e9527b9a Another minor change in the manual
6dcfdb1d4b Avoid infinite recursion in file_directory_p
6a1c03d65b More minor changes in the Emacs manual
479f51a63b Make tooltip code handle scenarios from Bug#30182 and Bug#30399
2c980ea613 Handle "gio monitor" in tramp-sh.el
32fb8c4c33 Avoid printing garbled error message from image.el
875cb835f0 Merge from origin/emacs-26
0276743672 Doc string fix in latin-alt.el
c9269af2e3 Minor improvements in package.texi
898a3e4784 Minor improvement in Emacs user manual
8f7d718499 * doc/lispref/anti.texi (Antinews): Replace 25.2 with 25.3.
989d0afce9 Improve documentation of mini-window resizing
923e4157ba More changes in the Emacs manual
05e8b4392b Fix iswitchb highlighting: don't mutate buffer name
6b183f85e0 * lisp/emacs-lisp/cl-macs.el (cl--sm-macroexpand): Handle lambda!
565adf2e90 Clarify that text-quoting-style doesn't affect *Info* (Bug#24519)
2dd273b985 Mention that shell quoting of % on w32 may fail (Bug#19350)
19fa6d561a Fix --no-build-details for w32 and macOS
979944c9b0 * doc/lispref/anti.texi (Antinews): Add some more antinews.
d34dbc0b69 Remove unused duplicate of minibuffer-completion-contents
c2727e3c40 Fix crashes when run with --no-build-details
dc08490ac7 More changes for the manual
04c5bd5b1f Doc fixes for process functions (Bug#30349)
e209034536 Fix shr and CSS 4 color maps
5fe81923e5 Yet another round of improvements in the manual
a710f8aa61 * lisp/comint.el (comint-history-isearch-setup): Check if process is live.
0372233906 * lisp/dired.el (dired-get-marked-files): Add new optional arg 'error'.
109237e23a ; Small fixes and indentation
31350817ae Support occur command operating on the region from Isearch.
5c414441ed Work around macOS faccessat bug
8e42b1bd3c Support list-matching-lines-jump-to-current-line for context lines.
a0c7157a16 Merge from origin/emacs-26
c24c5dc4a4 Fix inconsistency expanding "//" in Tramp
c787a49682 * lisp/vc/vc-git.el (vc-git-print-log): Restrict file scope to a single file
b654791064 * doc/emacs/misc.texi (Interactive Shell): Refer to node "Minibuffer File"
f1102d2de4 Yet another round of fixing the Emacs manual
8fbf28536e Fix handling of modifier keys on macOS
76b5a687d0 * etc/NEWS: Expunge the solecism "allow(s)" + infinitive
d2630e4569 Make tramp-archive fit for older Emacsen
b4ff8cc591 Two minor fixes in Antinews
aafcd12eeb * etc/NEWS: Rename image-dired-thumb-job-limit
a893924fd1 * lisp/simple.el (async-shell-command, shell-command): Fix grammar
699081f051 Fix deferred display of async shell-command buffers
d2d5e54824 Mention remote file name completion in Emacs manual
327d251f8a Avoid EOVERFLOW problems with file-directory-p
a34c7d7470 Add tests to verify error propagation in 'json-insert'.
9da8da2c41 Don't require all file-attributes to be equal (Bug#30327)
a057771a5a * test/lisp/files-tests.el (files-tests--with-temp-non-special):
a2cb52cd2e Prevent loading tramp-archive when it cannot be used
f7c8a12b12 ; Fix arglist doc of json parse functions
84c9dba4ce Autoload tramp-archive
f589f5ae6e Yest another round of manual copyedits
1ed408995a Update xdisp.c commentary
e23de39e22 Fix Bug#30324
e1a9dc0af6 Recognize Org as builtin package (bug#30310)
0e0ad865d4 ; * autogen.sh (check_version): Improve previous change.
c532a8015e Try to work around intermittent autogen failure on hydra.nixos.org
83fc9009c6 ; * autogen.sh: Remove temporary debug statements.
8298cb41fe ; * autogen.sh: Add some temporary debug statements.
1360df13af ; * autogen.sh: Remove temporary debug statements.
028087d903 ; * autogen.sh: Add some temporary debug statements.
396ed68739 * autogen.sh: Remove temporary debug statements.
b70321ea85 * autogen.sh: Add some temporary debug statements, for hydra.nixos.org.
2b288eb900 ; Merge from emacs-26
4c8157cfe0 ; Let files-tests.el pass multiple times in a session
99251ab62e ; Avoid process kill query during term tests
d3090a3a3e CC Mode: Fix an enum intro being parsed as defun-block-intro
0443411f5c Properly integrate modules into the loading process (Bug#30164).
75c663f834 Use 'defalias' in test module instead of 'fset'.
f3d0db7f0f Add support for module functions to C-h f (Bug#30163).
344750aef4 Handle quoted remote file names for file notifications
b86b8ee072 Minor tweaks in tramp-archive.el
a893a4db20 Merge from origin/emacs-26
8dd2edc70f ; Merge from origin/emacs-26
c4f8b5545e Merge from origin/emacs-26
6f495abd0d Another round of manual fixups
6ff8b7749f * lisp/emacs-lisp/cconv.el (cconv-convert): Fix compiling compiled code
59344c43a9 * doc/lispref/customize.texi (Custom Themes): Clarify .el preference.
ecc6257f48 Fix quoting in tramp-find-inline-compress for w32
6386efc8f7 * doc/emacs/entering.texi (Entering Emacs): Another wording fix.
e08de2bae2 Handle quoted file names in filenotify.el
00c65bcf4e Use file-name-quote{,d-p} in files-tests.el
65da409e41 Test and fix "/:" quoted file name handlers (Bug#30243)
e21f0189f3 * doc/lispref/functions.texi (Inline Functions): Fix typo (Bug#30238).
1c2fb04bee * lisp/imenu.el (imenu-generic-expression): Rephrase doc (Bug#30294).
b90e91ca54 Fix last change of @key markup
f180075f15 * doc/emacs/entering.texi (Entering Emacs): Fix markup.
100e33979b Update from gnulib
c85d9e401a * src/xfns.c: Make some non-exported vars static.
01932c8dec Revert a1bbc49015 (Bug#30243), do not merge
855ae578ab ; Auto-commit of loaddefs files.
56c37bbdb6 Use @key{} where it is missing in the manuals
25c01983af Another round of manual fixes from proofreading

### CODE REVIEW

**Comment by shaleh** [comment]
This gets us to Feb 14 2018.

**Comment by shaleh** [comment]
Most of the C changes are related to sharing docstrings when the DEFUN is #ifdef'ed between files. There is typically a term, X window, w32, etc. version of the same function. This allows the help text to be placed on just one of them, usually the X window version.

**Comment by shaleh** [comment]
@harryfei can I tag out and ask you to look at why the tramp test failed?

**Comment by harryfei** [comment]
@shaleh Yeah sure. I can dig into the problem.

**Comment by shaleh** [comment]
Thanks. 2am. Sleep time.

File: test/lisp/net/tramp-tests.el [hunk 1]
```diff
@@ -2993,32 +3013,23 @@ This tests also `make-symbolic-link', `file-truename' and `add-name-to-file'."
...
-	     tmp-name2)
+	    (make-symbolic-link "/penguin:motd:" tmp-name2)
 	    (should (file-symlink-p tmp-name2))
 	    (should
 	     (string-equal
 	      (file-truename tmp-name2)
 	      (tramp-compat-file-name-quote
 	       (concat (file-remote-p tmp-name2) "/penguin:motd:"))))
 	    ;; `tmp-name3' is a local file name.
-	    ;; `make-symbolic-link' might not be permitted on w32 systems.
```
[the remark this answers is not in the document]
Comment (shaleh) [reply test/lisp/net/tramp-tests.el]: Thanks. I must have chosen poorly on a merge.

**Comment by harryfei** [comment]
Did you merge in rebased mode?

**Comment by shaleh** [comment]
I squashed like a fool....

**Comment by harryfei** [comment]
I made the same mistake in my last merging because GitHub remembers my last merge strategy.

**Pull request merged**